pallets/amm/src/lib.rs

```diff
@@ -389,70 +399,78 @@ decl_module! {
 				Error::<T>::InsufficientPoolAssetBalance
 			);
 
-			// Note: this check is not really needed as we already check if amount to remove >= liquidity in pool
 			let liquidity_left = total_shares.checked_sub(liquidity_amount).ok_or(Error::<T>::InvalidLiquidityAmount)?;
 
-			T::Currency::transfer(asset_a, &pair_account, &who, remove_amount_a)?;
-			T::Currency::transfer(asset_b, &pair_account, &who, remove_amount_b)?;
+			with_transaction_result( || {
+				T::Currency::transfer(asset_a, &pair_account, &who, remove_amount_a)?;
+				T::Currency::transfer(asset_b, &pair_account, &who, remove_amount_b)?;
 
-			T::Currency::withdraw(share_token, &who, liquidity_amount)?;
+				T::Currency::withdraw(share_token, &who, liquidity_amount)?;
 
-			<TotalLiquidity<T>>::insert(&pair_account, liquidity_left);
+				<TotalLiquidity<T>>::insert(&pair_account, liquidity_left);
 
-			Self::deposit_event(RawEvent::RemoveLiquidity(who.clone(), asset_a, asset_b, liquidity_amount));
+				Self::deposit_event(RawEvent::RemoveLiquidity(who.clone(), asset_a, asset_b, liquidity_amount));
 
-			if liquidity_left == 0 {
-				<ShareToken<T>>::remove(&pair_account);
-				<PoolAssets<T>>::remove(&pair_account);
+				if liquidity_left == 0 {
+					<ShareToken<T>>::remove(&pair_account);
+					<PoolAssets<T>>::remove(&pair_account);
 
-				Self::deposit_event(RawEvent::PoolDestroyed(who, asset_a, asset_b));
-			}
+					Self::deposit_event(RawEvent::PoolDestroyed(who, asset_a, asset_b));
+				}
 
-			Ok(())
+				Ok(())
+			})
 		}
 
 		#[weight =  <T as Config>::WeightInfo::sell()]
 		pub fn sell(
 			origin,
-			asset_sell: AssetId,
-			asset_buy: AssetId,
-			amount_sell: Balance,
+			asset_in: AssetId,
+			asset_out: AssetId,
+			amount: Balance,
 			max_limit: Balance,
 			discount: bool,
 		) -> dispatch::DispatchResult {
 			let who = ensure_signed(origin)?;
 
-			<Self as AMM<_,_,_>>::sell(&who, asset_sell, asset_buy, amount_sell, max_limit, discount)
+			<Self as AMM<_,_,_,_>>::sell(&who, AssetPair{asset_in:asset_in, asset_out: asset_out}, amount, max_limit, discount)
```

Comment:
use shorthand assignment? {asset_in, asset_out} instead of {asset_in:asset_in, asset_out: asset_out}